### PR TITLE
Issue 1095 always sign outgoing encrypted messages

### LINKED ISSUE #1095: always sign outgoing encrypted messages
after #1058 - this simply means passing private key into the encryption function for signing

If the outgoing message (or file) is encrypted, it should also be signed.

### PR DESCRIPTION
**Pull request by DenBond7**
This PR added an opportunity to always sign outgoing encrypted messages

close #1095 // if this PR closes an issue

----------------------------------

**Tests** _(delete all except exactly one)_:
- Does not need tests (tested in PGPalinless lib)

--------------------------------

### To be filled by reviewers

I have reviewed that this PR... _(tick whichever items you personally focused on during this review)_:
- [x] addresses the issue it closes (if any)
- [x] code is readable and understandable
- [x] is accompanied with tests, or tests are not needed
- [x] is free of vulnerabilities

### CODE REVIEW

File: FlowCrypt/src/main/java/com/flowcrypt/email/api/email/EmailUtil.kt [hunk 1]
```diff
@@ -627,20 +629,37 @@ class EmailUtil {
...
+      val senderKeyDetails = SecurityUtils.getSenderKeyDetails(context, accountEntity, senderEmail)
+      var pubKeys: List<String>? = null
+      var prvKeys: List<String>? = null
+      var ringProtector: SecretKeyRingProtector? = null
+
+      if (outgoingMsgInfo.encryptionType === MessageEncryptionType.ENCRYPTED) {
+        val recipients = outgoingMsgInfo.getAllRecipients().toMutableList()
+        pubKeys = SecurityUtils.getRecipientsPubKeys(context, recipients)
+        pubKeys.add(senderKeyDetails.publicKey
+            ?: throw NullPointerException("Sender pub key not found"))
```
Comment (IvanPizhenko) [inline FlowCrypt/src/main/java/com/flowcrypt/email/api/email/EmailUtil.kt:650]: I think IllegalStateException is more appopriate

File: FlowCrypt/src/main/java/com/flowcrypt/email/api/email/EmailUtil.kt [hunk 2]
```diff
@@ -627,20 +629,37 @@ class EmailUtil {
...
+        prvKeys = listOf(senderKeyDetails.privateKey
+            ?: throw NullPointerException("Sender private key not found"))
```
Comment (IvanPizhenko) [inline FlowCrypt/src/main/java/com/flowcrypt/email/api/email/EmailUtil.kt:652]: I think IllegalStateException is more appopriate

File: FlowCrypt/src/main/java/com/flowcrypt/email/jetpack/workmanager/ForwardedAttachmentsDownloaderWorker.kt [hunk 3]
```diff
@@ -199,11 +199,17 @@ class ForwardedAttachmentsDownloaderWorker(context: Context, params: WorkerParam
...
+    var pubKeys: List<String>? = null
+
+    if (msgEntity.isEncrypted == true) {
+      val senderEmail = EmailUtil.getFirstAddressString(msgEntity.from)
+      val recipients = msgEntity.allRecipients.toMutableList()
+      pubKeys = SecurityUtils.getRecipientsPubKeys(applicationContext, recipients)
+      val senderKeyDetails = SecurityUtils.getSenderKeyDetails(applicationContext,
+          account, senderEmail)
+      pubKeys.add(senderKeyDetails.publicKey
+          ?: throw NullPointerException("Sender pub key not found"))
```
Comment (IvanPizhenko) [inline FlowCrypt/src/main/java/com/flowcrypt/email/jetpack/workmanager/ForwardedAttachmentsDownloaderWorker.kt:211]: I think IllegalStateException is more appopriate

**Review by IvanPizhenko — CHANGES_REQUESTED** [review]
Please change exception type in the commented below

File: FlowCrypt/src/main/java/com/flowcrypt/email/api/email/EmailUtil.kt (hunk 1 above)
Comment (DenBond7) [reply FlowCrypt/src/main/java/com/flowcrypt/email/api/email/EmailUtil.kt]: Agree, changed. Thanks!

File: FlowCrypt/src/main/java/com/flowcrypt/email/api/email/EmailUtil.kt (hunk 2 above)
Comment (DenBond7) [reply FlowCrypt/src/main/java/com/flowcrypt/email/api/email/EmailUtil.kt]: Changed

File: FlowCrypt/src/main/java/com/flowcrypt/email/jetpack/workmanager/ForwardedAttachmentsDownloaderWorker.kt (hunk 3 above)
Comment (DenBond7) [reply FlowCrypt/src/main/java/com/flowcrypt/email/jetpack/workmanager/ForwardedAttachmentsDownloaderWorker.kt]: Changed

**Review by seisvelas — APPROVED** [review]

**Review by IvanPizhenko — APPROVED** [review]

**Pull request merged**